### PR TITLE
Enables gradual, self-regulated learning by weighing updates based on model confidence.

### PR DESCRIPTION
**Pull request by thanay-sisir**
Enables gradual, self-regulated learning by weighing updates based on model confidence.

## Mechanism (Inverse Relationship)
* **High Certainty (Trusted Examples) → Low Learning Rate**
    * Protects existing knowledge from corruption.
    * Prevents large updates on already "known" data.
* **Low Certainty (Noisy/New Examples) → High Learning Rate**
    * Allows exploration and learning from uncertain or noisy information.

## Key Benefits
* Prevents overfitting to noise.
* Stabilizes training without external noise detection.
* Balances "protection of old knowledge" vs "learning of new info."

This is basically a fundamental principle that should be added long before but it is over-looked and missed. This is very very imp!!!!.....

### CODE REVIEW

**Comment by thanay-sisir** [comment]
@lukedarlow I really want you to look at this PR ASAP, where  I have included my explanation as part of the code luke.....

**Comment by lukedarlow** [comment]
This will not be accepted as a PR. Please, @thanay-sisir , try to understand the purpose of PRs, particularly for this repo: this is experimental code for the CTM that accompanies our release in May. It is not a scratchpad for research. 

If your changes in this PR are truly a "fundamental principle that should be added...", then I highly encourage you to run extensive experiments and prove this in the form of a **research paper**. That is the correct mechanism to go about something like this. Instead, you are making some changes and claiming them to be fundamental without any quantitative evidence to support this. 

Regarding this specific contribution/idea, I shall give you my feedback. The idea that the LR should be low for high certainty examples seems, on the surface, to be a good idea. But, it is not. Consider what happens to the **effective learning rate** in the standard setup, under the following conditions:

1. Correct and certain: very low loss, and therefore very low **effective learning rate**.
2. Incorrect and uncertain: high loss, and therefore very high **effective learning rate**.
3. Correct and uncertain: high loss, and therefore very high **effective learning rate**.
4. Incorrect and certain: high loss, and therefore very high **effective learning rate**.

What your propose essentially removes 4, lowering the learning rate in a circumstance where it is essentially. In other words, beyond surface level, your idea has fundamental flaws. 

-----------

I will now close this PR without acceptance. Should you wish to extend this conversation feel free to contact me via email, but I would very much appreciate it if you refrain from insisting on the acceptance of research-oriented PRs in the future.
